components/sup/src/manager/service/spec.rs

```diff
@@ -418,6 +422,7 @@ impl FromStr for ServiceSpec {
 pub struct ServiceBind {
     pub name: String,
     pub service_group: ServiceGroup,
+    /// Only set if this is a bind targeting a composite service
```

Comment:
Maybe call it `composite_service_name`, then?

Reply:
Or even better, change `ServiceBind` to make the difference explicit:
```rust
pub enum ServiceBind {
    Simple {
        name: String,
        service_group: ServiceGroup,
    },
    Composite {
        name: String,
        service_group: ServiceGroup,
        service_name: String,
    },
}
```
or
```rust
pub enum ServiceBindServiceName {
    Simple {},
    Composite { service_name: String },
}

pub struct ServiceBind {
    pub name: String,
    pub service_group: ServiceGroup,
    pub service_name: ServiceBindServiceName,
}
```
But that may be too disruptive at the current moment

Reply:
This is more or less exactly what I'd like to do, but didn't want to tackle it at this point because this is getting rather close to the protobuf representations, and I think we need to take another look at how we're doing the protobuf-type-to-Habitat-internal-Rust-type conversion there. I think we ultimately want to do something like we're doing with Butterfly, but that's gonna be it's own dedicated change.